src/wp-includes/class-wp-script-modules.php

```diff
@@ -949,4 +949,12 @@ public function print_a11y_script_module_html() {
 			. '<div id="a11y-speak-polite" class="a11y-speak-region" aria-live="polite" aria-relevant="additions text" aria-atomic="true"></div>'
 			. '</div>';
 	}
+
+	public function clone_without_enqueued_modules(): WP_Script_Modules {
+		$clone                                    = new WP_Script_Modules();
+		$clone->registered                        = $this->registered;
+		$clone->dependents_map                    = $this->dependents_map;
+		$clone->modules_with_missing_dependencies = $this->modules_with_missing_dependencies;
```

Comment:
Would this be better?

```suggestion
		$clone = clone $this;
		$clone->queue = array();
		$clone->done  = array();
```

Reply:
The result is the same, right? I don't mind either way, if you have reasons to prefer one or the other I'd be happy to know.

Reply:
Mutating the `queue` member variable is done elsewhere in core for `WP_Styles`. 

And this is just changing two members instead of three. And these two members feel less internal than the others, meaning we won't need to worry about ensuring other members get copied.

And if I understand correctly, resetting the `queue` and `done` are conceptually closer to what is desired: reset the state of what has been queued/printed to capture what is newly to be queued.